src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -81,12 +90,33 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        List<Person> personToEditList = lastShownList.stream()
+                .filter(x -> x.getIc().equals(nric))
+                .collect(Collectors.toList());
+
+        if (personToEditList.size() == 0) {
+            throw new CommandException(MESSAGE_DOESNT_EXIST);
         }
 
-        Person personToEdit = lastShownList.get(index.getZeroBased());
-        Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+        //developer assumption - can't have 2 people with same IC
+        assert personToEditList.size() < 2;
+
+        Person personToEdit = personToEditList.get(0);
+        Person editedPerson;
+
+        if (personToEdit instanceof Patient) {
```

Comment:
Possible to extract out this part of the function? Think the execute is a bit too long and will cost us readability.